pkgs/by-name/as/astro-language-server/package.nix

```diff
@@ -0,0 +1,104 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  stdenvNoCC,
+  jq,
+  moreutils,
+  nodePackages,
+  cacert,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "astro-language-server";
+  version = "2.9.0";
+
+  src = fetchFromGitHub {
+    owner = "withastro";
+    repo = "language-tools";
+    rev = "@astrojs/language-server@${finalAttrs.version}";
+    hash = "sha256-OfPe2bGcYWXZHkFc0iCiU3dPxbSsxX+ZVN6Fd7bMf9o=";
+  };
+
+  sourceRoot = "${finalAttrs.src.name}/packages/language-server";
+
+  pnpm-deps = stdenvNoCC.mkDerivation {
+    pname = "${finalAttrs.pname}-pnpm-deps";
+    inherit (finalAttrs) src version;
+
+    nativeBuildInputs = [
+      jq
+      moreutils
+      nodePackages.pnpm
+      cacert
+    ];
+
+    # https://github.com/NixOS/nixpkgs/blob/de80f1eeac3152c5bbfb1f8891b6414d526bfc54/pkgs/by-name/po/pot/package.nix#L54
+    installPhase = ''
+      export HOME=$(mktemp -d)
+      pnpm config set store-dir $out
+      pnpm config set side-effects-cache false
+      pnpm config set update-notifier false
+      # use --ignore-script and --no-optional to avoid downloading binaries
+      # use --frozen-lockfile to avoid checking git deps
+      pnpm install --frozen-lockfile --no-optional --ignore-script --force
+
+      # Remove timestamp and sort the json files
+      rm -rf $out/v3/tmp
+      for f in $(find $out -name "*.json"); do
+        sed -i -E -e 's/"checkedAt":[0-9]+,//g' $f
+        jq --sort-keys . $f | sponge $f
+      done
+    '';
+
+    dontConfigure = true;
+    dontFixup = true;
+    dontBuild = true;
+    outputHashMode = "recursive";
+    outputHash = "sha256-Y6CPjZZrIgMpu54511G3EXOuSfJ/nTS8fLYQmhQaT4g=";
+  };
+  nativeBuildInputs = [
+    nodePackages.pnpm
+    nodePackages.nodejs
+  ];
+
+  preBuild = ''
+    export HOME=$(mktemp -d)
+    pnpm config set store-dir ${finalAttrs.pnpm-deps}
+    pnpm config set recursive-install false
+    pnpm config set package-manager-strict false
+
+    chmod -R +w ../..
+
+    pnpm install --offline --frozen-lockfile --no-optional --ignore-script
```

Comment:
```suggestion
    pnpm install --offline --frozen-lockfile --ignore-script
```

Reply:
With this change, I'm now getting a different error(https://gist.github.com/pyrox0/ccff2fc03c7821427cc48b82ffd90df9#file-build2-log), and I still can't figure out what's going on there. Any ideas?